File: 2025-12/src/day_08/boj_15965/Main.cpp
#include <bits/stdc++.h>
using namespace std;

const int MAX = 7368787;
bool isPrime[1 + MAX];

void sieve() {
    fill(isPrime, isPrime + MAX + 1, true);
    isPrime[0] = isPrime[1] = false;

    for (int i = 2; i * i <= MAX; i++) {
        if (isPrime[i]) {
            for (int j = i * i; j <= MAX; j += i) {
                isPrime[j] = false;
            }
        }
    }
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int k;
    cin >> k;

    sieve();
    int cnt = 0;

    for (int i = 2; i <= MAX; i++) {
        if (isPrime[i]) cnt++;

        if (cnt == k) {
            cout << i;
            break;
        }
    }
}

File: 2025-12/src/day_08/boj_15965/Main.java
package day_08.boj_15965;

import java.io.*;
import java.util.*;

public class Main {

    static final int MAX = 7_368_787;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int k = Integer.parseInt(br.readLine());
        boolean[] isPrime = sieve();
        int cnt = 0;

        for (int i = 2; i <= MAX; i++) {
            if (isPrime[i]) cnt++;

            if (cnt == k) {
                System.out.println(i);
                break;
            }
        }
    }

    static boolean[] sieve() {
        boolean[] isPrime = new boolean[1 + MAX];
        Arrays.fill(isPrime, true);
        isPrime[0] = isPrime[1] = false;

        for (int i = 2; i * i <= MAX; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= MAX; j += i) {
                    isPrime[j] = false;
                }
            }
        }

        return isPrime;
    }
}
